contracts/Vault.sol

```diff
@@ -76,43 +77,49 @@ contract Vault is ExchangeRate, Initializable, OwnableUpgradeSafe {
     /// @notice Deposits `_amount` of `token` into Aave pool and mints equivalent bTokens. 
     /// @param _amount The amount to deposit.
     function deposit(uint256 _amount) public {
+        require(_amount > 0, "amount to deposit cannot be zero");
         _stateTransition();
 
         token.transferFrom(msg.sender, address(this), _amount);
         token.approve(address(aavePool), _amount);
+        uint256 beforeBalance = aToken.balanceOf(address(this));
         aavePool.deposit(
             address(token),
             _amount,
             address(this),
             0
         );
+        require(aToken.balanceOf(address(this)) >= beforeBalance, "atoken balance after deposit must be more than before");
```

Comment:
how do we test this requirement? or is it for debugging purposes to ensure that `aavePool.deposit` is working as we expect? Is this call passing in local tests but not in testnet?

Reply:
> is it for debugging purposes to ensure that aavePool.deposit is working as we expect

this, I often get a generic error from SafeMath ie. `subtraction overflow`, but there is no way to see  the stack trace and where it failed on non-Hardhat environment...